File: app/src/main/java/com/example/customerglu/CustomerGluManager.java
//package com.example.customerglu;
//import android.app.Activity;
//import android.content.BroadcastReceiver;
//import android.content.Context;
//import android.content.Intent;
//import android.content.IntentFilter;
//import android.os.Bundle;
//import android.os.Handler;
//import android.util.Log;
//import android.widget.Toast;
//
//import com.customerglu.sdk.CustomerGlu;
//import com.customerglu.sdk.Interface.CGDeepLinkListener;
//import com.customerglu.sdk.Interface.DataListner;
//import com.customerglu.sdk.Modal.DeepLinkWormholeModel;
//import com.customerglu.sdk.Modal.NudgeConfiguration;
//import com.customerglu.sdk.Modal.RegisterModal;
//import com.customerglu.sdk.Utils.CGConstants;
//
//import org.json.JSONException;
//import org.json.JSONObject;
//
//import java.util.HashMap;
//import java.util.Set;
//
//public class CustomerGluManager {
//
//    public static void initializeSDK(Context context, boolean debugMode) {
//        CustomerGlu.getInstance().initializeSdk(context);
//        CustomerGlu.getInstance().gluSDKDebuggingMode(context, debugMode);
//        setUpCTACallback(context);
//    }
//
//    private static void setUpCTACallback(Context context) {
//
//        BroadcastReceiver mMessageReceiver = new BroadcastReceiver() {
//            @Override
//            public void onReceive(Context context, Intent intent) {
//                try {
//                    new Handler().postDelayed(() -> {
//                        if (intent.getAction().equals("CUSTOMERGLU_DEEPLINK_EVENT")) {
//                            String data = intent.getStringExtra("data");
//                            try {
//                                JSONObject jsonObject = new JSONObject(data);
//                                if (jsonObject.has("deepLink")) {
//                                    String deeplink = jsonObject.getString("deepLink");
//                                    Toast.makeText(context, "Please handle CTA Deeplink", Toast.LENGTH_SHORT).show();
//                                }
//                            } catch (Exception e) {
//                                Log.e("CUSTOMERGLU",""+e);
//                            }
//                        }
//                        if (intent.getAction().equals("CUSTOMERGLU_ANALYTICS_EVENT")) {
//                            String data = intent.getStringExtra("data");
//                            try {
//                                JSONObject jsonObject = new JSONObject(data);
//                                Log.d("ANALYTICS_EVENT", data);
//
//                            } catch (Exception e) {
//                                Log.e("CUSTOMERGLU",""+e);
//
//                            }
//                            // Log.e("WebAnalytics ", "" + jsonObject);
//                            // Toast.makeText(context, "Analytics " + jsonObject, Toast.LENGTH_SHORT).show();
//                        }
//                        if (intent.getAction().equals("CUSTOMERGLU_BANNER_LOADED")) {
//                            String data = intent.getStringExtra("data");
//                            Log.d("BANNER_LOADED", data);
//
//                        }
//                        if (intent.getAction().equals("CG_INVALID_CAMPAIGN_ID")) {
//                            String data = intent.getStringExtra("data");
//                             Log.e(" INVALID CampaignId", data);
//                        }
//                        // Add the logic to redirect to appropriate page
//                    }, 500);
//                } catch (Exception e) {
//                    System.out.println(e);
//                }
//            }
//        };
//        context.registerReceiver(mMessageReceiver, new IntentFilter("CUSTOMERGLU_DEEPLINK_EVENT"));
//        context.registerReceiver(mMessageReceiver, new IntentFilter("CUSTOMERGLU_ANALYTICS_EVENT"));
//        context.registerReceiver(mMessageReceiver, new IntentFilter("CUSTOMERGLU_BANNER_LOADED"));
//        context.registerReceiver(mMessageReceiver, new IntentFilter("CG_INVALID_CAMPAIGN_ID"));
//    }
//
//    public static void enableEntryPoints(Context context) {
//        CustomerGlu.getInstance().enableEntryPoints(context, true);
//    }
//
//    public static void registerUser(Context context, HashMap<String, Object> userData) {
//        CustomerGlu.getInstance().registerDevice(context, userData, new DataListner() {
//            @Override
//            public void onSuccess(RegisterModal registerModal) {
//                // TODO
//            }
//
//            @Override
//            public void onFail(String message) {
//                // TODO
//            }
//        });
//    }
//
//    public static void clearGluData(Context context) {
//        CustomerGlu.getInstance().clearGluData(context);
//    }
//
//    public static void enableAnalytics() {
//        CustomerGlu.getInstance().enableAnalyticsEvent(true);
//    }
//
//    public static void openWallet(Context context, NudgeConfiguration nudgeConfiguration) {
//        CustomerGlu.getInstance().openWallet(context, nudgeConfiguration);
//    }
//
//    public static void loadCampaignById(Context context, String campaignId, NudgeConfiguration nudgeConfiguration) {
//        CustomerGlu.getInstance().loadCampaignById(context, campaignId, nudgeConfiguration);
//    }
//
//    public static void openNudge(Context context, String nudgeId, NudgeConfiguration nudgeConfiguration) {
//        CustomerGlu.getInstance().openNudge(context, nudgeId, nudgeConfiguration);
//    }
//
//    public static void setClassNameForCG(Activity activity, String screenName) {
//        CustomerGlu.getInstance().showEntryPoint(activity, screenName);
//    }
//    public void displayCGNotifications(Context context, JSONObject jsonObject, int icon) {
//        CustomerGlu.getInstance().displayCustomerGluNotification(context, jsonObject, icon);
//    }
//
//    public void displayCGBackgroundNotification(Activity activity, long delayInSecond) {
//        Intent parent_intent = activity.getIntent();
//        String intentSource = "none";
//        if (parent_intent != null) {
//            Bundle extras = parent_intent.getExtras();
//            if (extras != null) {
//                intentSource = extras.getString("type", "none");
//                String myType = extras.getString("from", "none");
//                if (intentSource.equalsIgnoreCase("CustomerGlu")) {
//                    JSONObject json = new JSONObject();
//                    Set<String> keys = extras.keySet();
//                    for (String key : keys) {
//                        try {
//                            json.put(key, JSONObject.wrap(extras.get(key)));
//                        } catch (JSONException e) {
//                            //Handle exception here
//                        }
//                    }
//                    new Handler().postDelayed(new Runnable() {
//                        @Override
//                        public void run() {
//                            CustomerGlu.getInstance()
//                                    .displayCustomerGluBackgroundNotification(activity.getApplicationContext(), json);
//                        }
//                    }, (delayInSecond * 100));
//                }
//            }
//        }
//    }
//
//    public void setDarkTheme(Context context, boolean value) {
//        CustomerGlu.getInstance().setDarkMode(context, value);
//    }
//
//    public void listenToSystemDarkMode(boolean value) {
//        CustomerGlu.getInstance().listenToSystemDarkMode(value);
//    }
//
//    public void setCGDeeplinkData(Activity activity) {
//        CustomerGlu.getInstance().setupCGDeepLinkIntentData(activity);
//    }
//
//    public void setCgDeeplinkCallbackListener() {
//        CustomerGlu.getInstance().setCgDeepLinkListener(new CGDeepLinkListener() {
//            @Override
//            public void onSuccess(CGConstants.CGSTATE message, DeepLinkWormholeModel.DeepLinkData deepLinkData) {
//                if (message == CGConstants.CGSTATE.DEEPLINK_URL) {
//                    String url = "";
//                    if (deepLinkData.getContent().getUrl() != null) {
//                        url = deepLinkData.getContent().getUrl();
//                    }
//                    // Add your logic
//                }
//                if (message == CGConstants.CGSTATE.SUCCESS) {
//                    //TODO: Not yet implemented
//                }
//            }
//
//            @Override
//            public void onFailure(CGConstants.CGSTATE exceptions) {
//                switch (exceptions) {
//                    case EXCEPTION:
//
//                        break;
//                    case NETWORK_EXCEPTION:
//
//                        break;
//                    case INVALID_CAMPAIGN:
//
//                        break;
//                    case CAMPAIGN_UNAVAILABLE:
//
//                        break;
//                    case INVALID_URL:
//
//                        break;
//                    case USER_NOT_SIGNED_IN:
//
//                        break;
//                    default:
//                        break;
//                }
//            }
//        });
//    }
//
//    public void configureLoaderColour(Context context, String color) {
//        CustomerGlu.getInstance().configureLoaderColour(context, color);
//    }
//
//    public void sendEventsToCG(Context context, String eventName, HashMap<String, Object> eventProperties) {
//        CustomerGlu.getInstance().sendEvent(context, eventName, eventProperties);
//    }
//
//    public void allowAnonymousUserRegistration(boolean value)
//    {
//        CustomerGlu.getInstance().allowAnonymousRegistration(value);
//    }
//    public void openWalletAsFallback(boolean value)
//    {
//        CustomerGlu.getInstance().openWalletAsFallback(value);
//    }
//
//}

File: app/src/main/java/com/example/customerglu/Utils/CustomerGluManager.kt
package com.example.customerglu.Utils

import android.app.Activity
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.Handler
import android.util.Log
import android.widget.Toast
import com.customerglu.sdk.CustomerGlu
import com.customerglu.sdk.Interface.CGDeepLinkListener
import com.customerglu.sdk.Interface.DataListner
import com.customerglu.sdk.Modal.DeepLinkWormholeModel.DeepLinkData
import com.customerglu.sdk.Modal.NudgeConfiguration
import com.customerglu.sdk.Modal.RegisterModal
import com.customerglu.sdk.Utils.CGConstants.CGSTATE
import com.example.customerglu.HomeActivity
import com.example.customerglu.Utils.CommonMethods.Companion.navigateToActivity
import org.json.JSONException
import org.json.JSONObject


object CustomerGluManager {

    lateinit var mMessageReceiver: BroadcastReceiver

    fun initializeSDK(context: Context,debugMode: Boolean = false)
    {
        CustomerGlu.getInstance().initializeSdk(context)
        CustomerGlu.getInstance().gluSDKDebuggingMode(context,debugMode)
        setUpCTACallback(context)

    }

    private fun setUpCTACallback(context: Context) {

        mMessageReceiver = object : BroadcastReceiver() {
            override fun onReceive(context: Context, intent: Intent) {
                // Extract data included in the Intent
                try {

                    Handler().postDelayed({
                        if (intent.action.equals("CUSTOMERGLU_DEEPLINK_EVENT", ignoreCase = true)) {
                            val data = intent.getStringExtra("data")
                            val jsonObject = JSONObject(data)
                            if (jsonObject.has("deepLink")) {

                                var deeplink = jsonObject.get("deepLink")
                              //  Toast.makeText(context, ""+deeplink, Toast.LENGTH_SHORT).show()
                                if (deeplink.equals("customerglu://profile")) {
                                    navigateToActivity(context,Constants.Profile)

                                } else if (deeplink.equals("customerglu://cart")) {
                                    navigateToActivity(context,Constants.Cart)

                                } else if(deeplink.equals("customerglu://wishlist"))
                                {
                                    navigateToActivity(context,Constants.Wishlist)

                                }else if(deeplink.equals("customerglu://categories"))
                                {
                                    navigateToActivity(context,Constants.Categories)
                                }

                                else {
                                    navigateToActivity(context,Constants.Home)

                                }
                            }
                        }

                        if (intent.action.equals(
                                "CUSTOMERGLU_ANALYTICS_EVENT",
                                ignoreCase = true
                            )
                        ) {
                            val data = intent.getStringExtra("data")
                            val jsonObject = JSONObject(data)
                               Log.e("Analytics ", "" + jsonObject);
                            //  Toast.makeText(context, "Analytics " + jsonObject, Toast.LENGTH_SHORT).show();
                        }
                        if (intent.action.equals("CUSTOMERGLU_BANNER_LOADED", ignoreCase = true)) {
                            val data = intent.getStringExtra("data")

                            //   Toast.makeText(getApplicationContext(), data, Toast.LENGTH_LONG).show();
                            // banner.setVisibility(GONE);
                        }

                        if (intent.action.equals("CG_INVALID_CAMPAIGN_ID", ignoreCase = true)) {
                            val data = intent.getStringExtra("data")
                            Log.e("WebAnalytics INVALID", data.toString());
                        }

                            // Add the logic to redirect to appropriate page

                    }, 500)
                } catch (e: Exception) {
                    println(e)
                }
            }
        }

        context.registerReceiver(mMessageReceiver, IntentFilter("CUSTOMERGLU_DEEPLINK_EVENT"))
        context.registerReceiver(mMessageReceiver, IntentFilter("CUSTOMERGLU_ANALYTICS_EVENT"))
        context.registerReceiver(mMessageReceiver, IntentFilter("CUSTOMERGLU_BANNER_LOADED"))
        context.registerReceiver(mMessageReceiver, IntentFilter("CG_INVALID_CAMPAIGN_ID"))

    }

    fun enableEntryPoints(context: Context){
        CustomerGlu.getInstance().enableEntryPoints(context,true)
    }

    fun registerUser(context: Context,userData:HashMap<String,Any>)
    {
        CustomerGlu.getInstance().registerDevice(context,userData,object : DataListner {

            override fun onSuccess(registerModal: RegisterModal?) {
                val intent = Intent(context, HomeActivity::class.java)
                intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK)
                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
                // toast("Registered")
                if (registerModal != null) {
                    Prefs.putKey(
                        context,
                        "userId",
                        registerModal.data.getUser().userId
                    )
                }

               context.startActivity(intent)            }

            override fun onFail(message: String?) {
                 println(message)
            }
        })
    }



    fun logoutFromCG(context: Context)
    {
        CustomerGlu.getInstance().clearGluData(context)
    }

    fun enableAnalytics()
    {
        CustomerGlu.getInstance().enableAnalyticsEvent(true);
    }


    fun openWallet(context: Context,nudgeConfiguration: NudgeConfiguration = NudgeConfiguration())
    {
        CustomerGlu.getInstance().openWallet(context,nudgeConfiguration)
    }

    fun loadCampaignById(context: Context,campaignId:String,nudgeConfiguration: NudgeConfiguration = NudgeConfiguration())
    {
        CustomerGlu.getInstance().loadCampaignById(context,campaignId,nudgeConfiguration)
    }

    fun openNudge(context: Context,nudgeId:String,nudgeConfiguration: NudgeConfiguration = NudgeConfiguration())
    {
        CustomerGlu.getInstance().openNudge(context,nudgeId,nudgeConfiguration)
    }


    fun setClassNameForCG(activity: Activity,screenName:String)
    {
        CustomerGlu.getInstance().showEntryPoint(activity,screenName)
    }


    fun displayCGNotifications(context: Context,jsonObject:JSONObject,icon:Int)
    {
        CustomerGlu.getInstance()
            .displayCustomerGluNotification(context, jsonObject, icon)
    }

    fun displayCGBackgroundNotification(activity: Activity,delayInSecond:Long = 2)
    {
        val parent_intent = activity.intent
        var intentSource = "none"
        if (parent_intent != null) {
            if (parent_intent.extras != null) {
                intentSource = parent_intent.extras!!.getString("type", "none")
                val myType = parent_intent.extras!!.getString("from", "none")
                if (intentSource.equals("CustomerGlu", ignoreCase = true)) {
                    val data = parent_intent.extras
                    val json = JSONObject()
                    val keys = data!!.keySet()
                    for (key in keys) {
                        try {
                            json.put(key, JSONObject.wrap(data!![key]))
                        } catch (e: JSONException) {
                            //Handle exception here
                        }
                    }
                    Handler().postDelayed({
                        CustomerGlu.getInstance()
                            .displayCustomerGluBackgroundNotification(activity.applicationContext, json)
                    },(delayInSecond * 100))

                }
            }
        }
    }

    fun setDarkTheme(context: Context,value:Boolean)
    {
        CustomerGlu.getInstance().setDarkMode(context,value)
    }

    fun listenToSystemDarkMode(value: Boolean)
    {
        CustomerGlu.getInstance().listenToSystemDarkMode(value)
    }

    fun  setCGDeeplinkData(activity: Activity)
    {
        CustomerGlu.getInstance().setupCGDeepLinkIntentData(activity)
    }

    fun setCgDeeplinkCallbackListener()
    {
        CustomerGlu.getInstance().setCgDeepLinkListener(object : CGDeepLinkListener {
            override fun onSuccess(message: CGSTATE, deepLinkData: DeepLinkData) {
                if (message == CGSTATE.DEEPLINK_URL) {
                    var url: String? = ""
                    if (deepLinkData.content.url != null) {
                        url = deepLinkData.content.url
                    }
                    // Add your logic
                }

                if (message == CGSTATE.SUCCESS)
                {
                    println("")

                }

            }

            override fun onFailure(exceptions: CGSTATE) {

                when (exceptions)
                {
                    CGSTATE.EXCEPTION ->  println("")
                    CGSTATE.NETWORK_EXCEPTION ->    println("")
                    CGSTATE.INVALID_CAMPAIGN ->    println("")
                    CGSTATE.CAMPAIGN_UNAVAILABLE ->    println("")
                    CGSTATE.INVALID_URL ->    println("")
                    CGSTATE.USER_NOT_SIGNED_IN ->    println("")

                    else -> {}
                }

            }
        })
    }

    fun configureLoaderColour(context: Context,color:String)
    {
        CustomerGlu.getInstance().configureLoaderColour(context,color)
    }

    fun sendEventsToCG(context: Context,eventName:String,eventProperties:HashMap<String,Any>)
    {
        CustomerGlu.getInstance().sendEvent(context,eventName,eventProperties)
    }

    fun allowAnonymousUserRegistration(value: Boolean)
    {
        CustomerGlu.getInstance().allowAnonymousRegistration(value)
    }

    fun openWalletAsFallback(value: Boolean)
    {
        CustomerGlu.getInstance().openWalletAsFallback(value)
    }

}